Date: Mon, 31 Oct 2005 22:10:59 +0900
From: Horms <horms@verge.net.au>
To: linuxppc-dev@ozlabs.org
Cc: Andrew Morton <akpm@osdl.org>, david@woodhou.se
Subject: audit_sysctl_exit can only be used with CONF_AUDIT_SYSCTL

This section of code calls .audit_syscal_exit,
but is inside CONFIG_AUDIT, so it will fail to build
if CONFIG_AUDITSYSCALL is not defined. 

After discussion with David Woodhouse, change the 
ifdef to CONFIG_AUDITSYSCALL


Signed-off-by: Horms <horms@verge.net.au>

--
Backported to 2.6.14 by Horms

diff --git a/arch/ppc64/kernel/entry.S b/arch/ppc64/kernel/entry.S
index 984a106..2d22bf0 100644
--- a/arch/ppc64/kernel/entry.S
+++ b/arch/ppc64/kernel/entry.S
@@ -279,7 +279,7 @@ _GLOBAL(ppc32_rt_sigsuspend)
 	bne	syscall_exit
 	/* If sigsuspend() returns zero, we are going into a signal handler. We
 	   may need to call audit_syscall_exit() to mark the exit from sigsuspend() */
-#ifdef CONFIG_AUDIT
+#ifdef CONFIG_AUDITSYSCALL
 	ld	r3,PACACURRENT(r13)
 	ld	r4,AUDITCONTEXT(r3)
 	cmpdi	0,r4,0
